### PR TITLE
Reports

### PR DESCRIPTION
**Pull request by titoOdUA**
@alanvardy how do you like this approach?

Basically Report module would be responsible for simple reports like those done today / done yesterday.
For now, I added 3 variants to try it out.
I think that we can build on this skeleton further by making interactable report variants, applying some sorting methods, etc. (but we will need to decide how we would provide arguments for it)

For now, I added done_yesterday, done_today, and due_today (I run out of good chars for short flags, maybe you can help me here =) ).

Would like to hear your opinion and suggestions, if it looks ok would write some tests after the initial review.

I really dislike if else forest in task list fn, but I think I can refactor it later.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -113,7 +114,10 @@ fn cmd() -> Command {
                        Command::new("list").about("List all tasks in a project")
                          .arg(config_arg())
                          .arg(project_arg())
-                         .arg(flag_arg("scheduled", 's',  "Only list tasks that are scheduled for today and have a time")),
+                         .arg(flag_arg("scheduled", 's',  "Only list tasks that are scheduled for today and have a time"))
+                         .arg(flag_arg("done-today", 't',  "Only list tasks that were done today"))
+                         .arg(flag_arg("due-today", 'x',  "Only list tasks that are due today"))
+                         .arg(flag_arg("done-yesterday", 'd',  "Only list tasks that were done yesterday")),
```
Comment (alanvardy) [inline src/main.rs:120]: Running out of flags sucks! I don't have any strong opinions about the character choices and don't mind if we want to break some later.

**Review by alanvardy — COMMENTED** [review]
I'm a fan of what I am seeing here, I like it that you are leaning into types and structs, so it will be easy to refactor in the future as we add more functionality.

I'll push to have the expects removed from the code in the final review 👍

File: src/report.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    pub fn new(config: Config, project: &str, report_type: CommonReports) -> Self {
+        match report_type {
+            CommonReports::DoneYesterday => Report::form_done_yesterday_report(config, project),
+            CommonReports::DoneToday => Report::form_done_today_report(config, project),
+            CommonReports::DueToday => Report::form_due_today_report(config, project),
+        }
+    }
+
+    fn form_done_yesterday_report(config: Config, project: &str) -> Self {
+        let project_id = projects::project_id(&config, project).expect("Failed to get project ID");
```
[the remark this answers is not in the document]
Comment (titoOdUA) [reply src/report.rs]: yeah, fixed it, changed signature of functions a little

File: src/main.rs (hunk 1 above)
Comment (titoOdUA) [reply src/main.rs:120]: For now I just left it as it is, I just don't see the options.
Will rework in future, maybe will change some clap code

File: src/report.rs [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    pub project_name: String,
+    pub items: Vec<Item>,
+    pub config: Config,
+    pub report_type: CommonReports,
+}
+
+impl Report {
+    pub fn new(config: Config, project: &str, report_type: CommonReports) -> Result<Self, String> {
+        let project_id =
+            projects::project_id(&config, project).map_err(|_| format!("Failed to get project"))?;
```
Comment (alanvardy) [inline src/report.rs:37]: `map_err` could be skipped if we make the error from `project_id` clearer

**Review by alanvardy — COMMENTED** [review]
Looks good, just needs clippy errors resolved and a couple more tests

**Comment by titoOdUA** [comment]
Hi, unfortunately, I don't have time to work on it at the moment.
You can complete it if you want, I don't mind =)

I, probably, wouldn't be active for a week or two. Have a lot on my plate right now.

**Comment by titoOdUA** [comment]
Still need a week or two. But I have a little bit of time to occasionally do a review/fix bug, for example.

@alanvardy 
When I finally would have time to contribute, how do you like the idea to create some kind of chat in Discord where we can discuss the tod project? Direction, goals, etc.

**Comment by alanvardy** [comment]
That is a really interesting idea, I would prefer to not use Discord, only because I am not currently on there. Would Slack suffice? I can set up a free plan and invite you and @stacksjb if y'all would like me to. I understand that sometimes just being able to chat in realtime helps move conversations along.

Whatever doesn't need to be realtime-ish will hopefully stay on GitHub though so that there is a single source of truth for people coming to the repo.

**Comment by alanvardy** [comment]
Closing due to inactivity, feel free to reopen when you are ready!

**Pull request closed without merging**